etc/esp-idf/CMakeLists.txt

```diff
@@ -0,0 +1,162 @@
+    $ENV{OPENMRNPATH}/src/openlcb/StreamTransport.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TcpDefs.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TractionCvSpace.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TractionDefs.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TractionProxy.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TractionTestTrain.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TractionThrottle.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/TractionTrain.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/Velocity.cxx
+    $ENV{OPENMRNPATH}/src/openlcb/WriteHelper.cxx
+
+    
+    $ENV{OPENMRNPATH}/src/os/FakeClock.cxx
+    $ENV{OPENMRNPATH}/src/os/logging_malloc.cxx
+    $ENV{OPENMRNPATH}/src/os/MDNS.cxx
+    $ENV{OPENMRNPATH}/src/os/os.c
+    $ENV{OPENMRNPATH}/src/os/OSImpl.cxx
+    $ENV{OPENMRNPATH}/src/os/OSSelectWakeup.cxx
+    $ENV{OPENMRNPATH}/src/os/stack_malloc.c
+    $ENV{OPENMRNPATH}/src/os/TempFile.cxx
+    $ENV{OPENMRNPATH}/src/os/watchdog.c
+
+    $ENV{OPENMRNPATH}/src/utils/Base64.cxx
+    $ENV{OPENMRNPATH}/src/utils/Blinker.cxx
+    $ENV{OPENMRNPATH}/src/utils/Buffer.cxx
+    $ENV{OPENMRNPATH}/src/utils/CanIf.cxx
+    $ENV{OPENMRNPATH}/src/utils/ClientConnection.cxx
+    $ENV{OPENMRNPATH}/src/utils/ConfigUpdateListener.cxx
+    $ENV{OPENMRNPATH}/src/utils/constants.cxx
+    $ENV{OPENMRNPATH}/src/utils/Crc.cxx
+    $ENV{OPENMRNPATH}/src/utils/DirectHub.cxx
+    $ENV{OPENMRNPATH}/src/utils/DirectHubGc.cxx
+    $ENV{OPENMRNPATH}/src/utils/DirectHubLegacy.cxx
+    $ENV{OPENMRNPATH}/src/utils/errno_exit.c
+    $ENV{OPENMRNPATH}/src/utils/FdUtils.cxx
+    $ENV{OPENMRNPATH}/src/utils/FileUtils.cxx
+    $ENV{OPENMRNPATH}/src/utils/format_utils.cxx
+    $ENV{OPENMRNPATH}/src/utils/ForwardAllocator.cxx
+    $ENV{OPENMRNPATH}/src/utils/GcStreamParser.cxx
+    $ENV{OPENMRNPATH}/src/utils/GcTcpHub.cxx
+    $ENV{OPENMRNPATH}/src/utils/gc_format.cxx
+    $ENV{OPENMRNPATH}/src/utils/GridConnect.cxx
+    $ENV{OPENMRNPATH}/src/utils/GridConnectHub.cxx
+    $ENV{OPENMRNPATH}/src/utils/HubDevice.cxx
+    $ENV{OPENMRNPATH}/src/utils/HubDeviceSelect.cxx
+    $ENV{OPENMRNPATH}/src/utils/ieeehalfprecision.c
+    $ENV{OPENMRNPATH}/src/utils/JSHubPort.cxx
+    $ENV{OPENMRNPATH}/src/utils/logging.cxx
+    $ENV{OPENMRNPATH}/src/utils/Queue.cxx
+    $ENV{OPENMRNPATH}/src/utils/ReflashBootloader.cxx
+    $ENV{OPENMRNPATH}/src/utils/ServiceLocator.cxx
+    $ENV{OPENMRNPATH}/src/utils/SocketCan.cxx
+    $ENV{OPENMRNPATH}/src/utils/SocketClient.cxx
+    $ENV{OPENMRNPATH}/src/utils/socket_listener.cxx
+    $ENV{OPENMRNPATH}/src/utils/Stats.cxx
+    $ENV{OPENMRNPATH}/src/utils/StringPrintf.cxx
+)
+
+set(INCLUDE_DIRS
+    $ENV{OPENMRNPATH}/src
+    $ENV{OPENMRNPATH}/include/esp-idf
+    $ENV{OPENMRNPATH}/include
+)
+
+if(CONFIG_ENABLE_ARDUINO_DEPENDS)
+    list(APPEND SOURCE_DIRS src)
+endif()
+
+set(IDF_DEPS
+    app_update
+    bootloader_support
+    driver
+    esp_adc
+    esp_app_format
+    esp_system
+    esp_wifi
+    espcoredump
+    hal
+    heap
+    vfs
+    mdns
+    bt)
```

Comment:
please add pthread since it is used in OSSelectWakeup for VFS registration.

Reply:
Done.